### DISCUSSION TITLE
DataLoader vs DataListLoader

### DISCUSSION THREAD

**Discussion by mdanb** [pyg-team/pytorch_geometric#2485] [Q&A]
The docs mention that DataListLoader "should be used for multi-gpu support via torch_geometric.nn.DataParallel." Why exactly is that?

**Comment by rusty1s** [comment]
Yes, this is a restriction since the `DataParallel` module will take care of splitting the batches to multiple GPUs, and therefore needs access to each individual example. A more elegant way is provided via `DistributedDataParallel`, where you can simply set up a data loader for each process/GPU, see [here](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/multi_gpu/distributed_batching.py).
